integration/update_test.go

```diff
@@ -770,6 +770,66 @@ func TestUpdateMany(t *testing.T) {
 	}
 }
 
+func TestMultiFlag(t *testing.T) {
+	t.Parallel()
+
+	for name, tc := range map[string]struct {
+		filter bson.D
+		update bson.D
+		multi  string
+		stat   bson.D
+	}{
+		"MultiFalse": {
+			filter: bson.D{{"foo", "x"}},
+			update: bson.D{{"$set", bson.D{{"foo", "y"}}}},
+			multi:  "false",
+			stat:   bson.D{{"n", int32(1)}, {"nModified", int32(1)}, {"ok", float64(1)}},
+		},
+		"MultiTrue": {
+			filter: bson.D{{"foo", "x"}},
+			update: bson.D{{"$set", bson.D{{"foo", "y"}}}},
+			multi:  "true",
+			stat:   bson.D{{"n", int32(2)}, {"nModified", int32(2)}, {"ok", float64(1)}},
+		},
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			ctx, collection := setup(t)
+
+			_, err := collection.InsertMany(ctx, []any{
+				bson.D{{"_id", "first"}, {"foo", "x"}},
+				bson.D{{"_id", "second"}, {"foo", "x"}},
+			})
+			require.NoError(t, err)
+
+			var actual bson.D
+
+			if tc.multi == "true" {
+				collection.UpdateMany(ctx, tc.filter, tc.update)
+			} else {
+				collection.UpdateOne(ctx, tc.filter, tc.update)
+			}
+
+			err = collection.FindOne(ctx, bson.D{{"_id", "first"}}).Decode(&actual)
+			require.NoError(t, err)
+
+			require.Equal(t, bson.D{{"_id", "first"}, {"foo", "y"}}, actual)
```

Comment:
Could you make a field `res` of type `bson.D` in the test table and move the expected documents for each test there? It allows for better readability (you see the query and what you get out of it), and also it would be consistent with already existing `TestUpdate/.../` functions

Reply:
(Also it would be easier to add new tests cases if needed)